src/pages/index.js

```diff
@@ -100,15 +86,356 @@ const cardSection = new Section(
   ".cards__list"
 );
 
-cardSection.renderItems();
-
-function createCard(cardData) {
-  return new Card(cardData, "#card-template", handleImageClick).generateCard();
+function loadInitialCards() {
+  api.getInitialCards()
+    .then((cards) => {
+      console.log("Fetched cards:", cards);
+      cardSection.renderItems(cards);
+    })
+    .catch((error) => {
+      console.error("Error loading initial cards:", error);
+    });
 }
+  loadInitialCards();
+  
+  function createCard(cardData) {
+    console.log("Creating card with data:", cardData);
+    return new Card(
+      {
+        name: cardData.name,
+        link: cardData.link,
+        id: cardData._id,
+        isLiked: cardData.isLiked
+      },
+      "#card-template",
+      handleImageClick,
+      handleDeleteCard,
+      handleLikeIcon
+    ).generateCard();
+  }
+  /* -------------------------------------------------------------------------- */
+  /*                               Popup Instances                              */
+  /* -------------------------------------------------------------------------- */
+  
+  const imagePopup = new PopupWithImage("#preview-card-modal");
+  imagePopup.setEventListeners();
+  
+  const profileFormPopup = new PopupWithForm(
+    "#edit-profile-modal",
+    handleProfileFormSubmit
+  );
+  profileFormPopup.setEventListeners();
+  
+  const addCardFormPopup = new PopupWithForm(
+    "#add-card-modal",
+    handleAddCardFormSubmit
+  );
+  addCardFormPopup.setEventListeners();
+  
+  const deleteConfirm = new PopupWithConfirm("#delete-card-modal");
+  deleteConfirm.setEventListeners();
+  
+  const avatarPopup = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+  avatarPopup.setEventListeners();
+  
+/* -------------------------------------------------------------------------- */
+/*                               Event Handlers                               */
+/* -------------------------------------------------------------------------- */
 
 function handleImageClick(name, link) {
-    imagePopup.open(name, link);
+  imagePopup.open(name, link);
 }
 
+function handleProfileFormSubmit(formData) {
+  const profileFormSubmitButton = document.querySelector("#profile-edit-submit-button");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine